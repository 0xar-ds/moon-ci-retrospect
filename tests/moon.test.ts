import * as path from "node:path";
import { $ } from "execa";
import { expect, test } from "vitest";

const indexJs = path.resolve("dist/index.js");

test("basic", async () => {
  const cwd = path.join(import.meta.dirname, "workspaces/basic");
  const { stdout, stderr } = await $({ cwd })`node ${indexJs}`;
  expect(stdout).toMatchSnapshot();
  expect(stderr).toBe("");
});

test("no-tasks", async () => {
  const cwd = path.join(import.meta.dirname, "workspaces/no-tasks");
  const { stdout, stderr } = await $({ cwd })`node ${indexJs}`;
  expect(stdout).toMatchSnapshot();
  expect(stderr).toBe("");
});
