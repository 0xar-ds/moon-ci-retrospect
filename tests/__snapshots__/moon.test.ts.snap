// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`basic 1`] = `
"::group::[42m PASS [49m [1mc:bar[22m
[34m$ echo bar1 && echo bar2 > /dev/stderr[39m
[48;5;236m„ÄÄ[32m‚è∫[39m STDOUT„ÄÄ[49m
bar1
[48;5;236m„ÄÄ[31m‚è∫[39m STDERR„ÄÄ[49m
bar2
::endgroup::
::group::[42m PASS [49m [1mc:baz[22m
[34m$ echo baz1 && echo baz2 > /dev/stderr[39m
[48;5;236m„ÄÄ[32m‚è∫[39m STDOUT„ÄÄ[49m
baz1
[48;5;236m„ÄÄ[31m‚è∫[39m STDERR„ÄÄ[49m
baz2
::endgroup::
::group::[42m PASS [49m [1mc:foo[22m
[34m$ echo foo1 && echo foo2 > /dev/stderr[39m
[48;5;236m„ÄÄ[32m‚è∫[39m STDOUT„ÄÄ[49m
foo1
[48;5;236m„ÄÄ[31m‚è∫[39m STDERR„ÄÄ[49m
foo2
::endgroup::
::group::[41m FAIL [49m [1mc:make-error[22m
[48;5;236m„ÄÄ[31m‚è∫[39m STDERR„ÄÄ[49m
This is an error message
::endgroup::
::group::[42m PASS [49m [1mb:bar[22m
[34m$ echo bar1 && echo bar2 > /dev/stderr[39m
[48;5;236m„ÄÄ[32m‚è∫[39m STDOUT„ÄÄ[49m
bar1
[48;5;236m„ÄÄ[31m‚è∫[39m STDERR„ÄÄ[49m
bar2
::endgroup::
::group::[42m PASS [49m [1mb:baz[22m
[34m$ echo baz1 && echo baz2 > /dev/stderr[39m
[48;5;236m„ÄÄ[32m‚è∫[39m STDOUT„ÄÄ[49m
baz1
[48;5;236m„ÄÄ[31m‚è∫[39m STDERR„ÄÄ[49m
baz2
::endgroup::
::group::[42m PASS [49m [1mb:foo[22m
[34m$ echo foo1 && echo foo2 > /dev/stderr[39m
[48;5;236m„ÄÄ[32m‚è∫[39m STDOUT„ÄÄ[49m
foo1
[48;5;236m„ÄÄ[31m‚è∫[39m STDERR„ÄÄ[49m
foo2
::endgroup::
::group::[41m FAIL [49m [1mb:make-error[22m
[48;5;236m„ÄÄ[31m‚è∫[39m STDERR„ÄÄ[49m
This is an error message
::endgroup::
::group::[42m PASS [49m [1ma:bar[22m
[34m$ echo bar1 && echo bar2 > /dev/stderr[39m
[48;5;236m„ÄÄ[32m‚è∫[39m STDOUT„ÄÄ[49m
bar1
[48;5;236m„ÄÄ[31m‚è∫[39m STDERR„ÄÄ[49m
bar2
::endgroup::
::group::[42m PASS [49m [1ma:baz[22m
[34m$ echo baz1 && echo baz2 > /dev/stderr[39m
[48;5;236m„ÄÄ[32m‚è∫[39m STDOUT„ÄÄ[49m
baz1
[48;5;236m„ÄÄ[31m‚è∫[39m STDERR„ÄÄ[49m
baz2
::endgroup::
::group::[42m PASS [49m [1ma:foo[22m
[34m$ echo foo1 && echo foo2 > /dev/stderr[39m
[48;5;236m„ÄÄ[32m‚è∫[39m STDOUT„ÄÄ[49m
foo1
[48;5;236m„ÄÄ[31m‚è∫[39m STDERR„ÄÄ[49m
foo2
::endgroup::
::group::[41m FAIL [49m [1ma:make-error[22m
[48;5;236m„ÄÄ[31m‚è∫[39m STDERR„ÄÄ[49m
This is an error message
::endgroup::"
`;

exports[`no-tasks 1`] = `"CI report file does not exist. No CI tasks may have been executed."`;
